### PR TITLE
Added D (under dlang) and Fixed C++ Error

### PR DESCRIPTION
**Pull request by Capital-EX**
I managed to make a port of bunnymark for D.  Additionally, I change `owner->remove_child(bunny)` to `bunnies->remove_child(bunny)` in the BunnymarkV2.cpp.   Although, I couldn't figure out what was up with BunnymarkV2.cs.

### My Benchmarks for D:
* Draw Texture:   15100
* Sprites: 10500
* V2: 8720

### CODE REVIEW

File: benchmarks/BunnymarkV2/cpp/BunnymarkV2.cpp [hunk 1]
```diff
@@ -102,7 +102,7 @@ class BunnymarkV2 : public GodotScript<Node2D> {
                 
                 Sprite* bunny = (Sprite*)bunnies->get_child(child_count - 1);
                 speeds.pop_back();
-                owner->remove_child(bunny);
```
Comment (cart) [inline benchmarks/BunnymarkV2/cpp/BunnymarkV2.cpp:105]: Bawhaha thats embarrassing for me

**Comment by cart** [comment]
Nice work!  I also couldn't figure out the V2.cs issue.  I think it might be a lifetime problem with GetChildren().  Maybe @neikeq knows. 

@neikeq: any insights?  we're getting this exception in BenchmarkV2.cs where we wouldn't expect one:

```
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at BunnymarkV2._Process (System.Single delta) [0x0004c] in /home/carter/Code/godot3-bunnymark/benchmarks/BunnymarkV2/cs/BunnymarkV2.cs:31 
```

**Comment by neikeq** [comment]
@cart Are all children of type Sprite?

**Comment by Capital-EX** [comment]
I've tried finding the offending value but It never shows up.  This might be an issue with Godot, because the error even appears when doing a type check:
```C#
if (bunnyChildren == null)  
    Console.WriteLine("BunnyChildren is null??"); // Not Called

else if (bunnyChildren[i] == null)
    Console.WriteLine($"Index {i} is null"); // Not Called

else if (!(bunnyChildren[i] is Sprite)) // This line will raise a System.NullReferenceException
    Console.WriteLine($"Index {i} is not a Sprite");
```
Additionally it seems multiple updates were made to GD Mono.

**Comment by cart** [comment]
@neikeq:  all children should be of type Sprite.  I created the member ```bunnies = new Node2D()``` and the only AddChild call is ```bunnies.AddChild(bunny)```, where bunny is of type Sprite.  But as @Capital-EX pointed out its pretty hard to verify that its actually true.

**Comment by Capital-EX** [comment]
@cart @neikeq 
After some poking a round it seems that an object *at random* will not be of type Sprite.  However, this `bad` object has some nastier properties:
* `bad.GetType() == typeof(Sprite)` will be false for this object.
* Any usage of `is` or `as` will throw a NullReferenceException
* Any attempt to access the properties/methods of it's type will throw a NullReferenceException.

So given all that I thought the type was some how null. However, `bad.GetType() == null` will return ***false***.

**Comment by cart** [comment]
@Capital-EX  I couldn't reproduce your first bullet but i could reproduce the others.  A bit more data:

1) bunnyChildren[i].GetType() == typeof(Sprite) is true for both "good" and "bad" objects
2) bunnyChildren[i].ToString() is "Godot.Sprite" for "good" objects
3) bunnyChildren[i].ToString() is "False" for "bad" objects.

**Comment by neikeq** [comment]
The issue should be fixed by godotengine/godot#12546.

**Comment by Capital-EX** [comment]
@neikeq Nice work 👍

**Pull request merged**